.github/workflows/frontend-docs-generation.yml

```diff
@@ -1,31 +1,37 @@
 name: Generate Documentation for the frontend
 
 on:
-    push:
-        branches:
-            - '**'
-    pull_request:
-        branches:
-            - '**'
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
 
 jobs:
-    generate-docs:
-        runs-on: ubuntu-latest
-        steps:
-            - name: Clone project
-              uses: actions/checkout@v3
+  generate-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v4
```

Comment:
Ensures full history for accurate dependency installation if needed ie: dept 0, please check @stephane-segning if it going to increase the actions time then we can remove that

Reply:
No need, as we focus only on the latest :)

Reply:
okay